sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -897,6 +898,24 @@ object SparkSession extends Logging {
       this
     }
 
+    // These configurations related to driver when deploy like `spark.master`,
+    // `spark.driver.memory`, this kind of properties may not be affected when
+    // setting programmatically through SparkConf in runtime, or the behavior is
+    // depending on which cluster manager and deploy mode you choose, so it would
+    // be suggested to set through configuration file or spark-submit command line options.
```

Comment:
@AngersZhuuuu, I would first document this explicitly and what happen for each configuration before taking an action to show a warning. Also, shouldn't we do this in `SparkContext`?

Reply:
We can document first, and then link the documentation URL in the warning message.

Reply:
> @AngersZhuuuu, I would first document this explicitly and what happen for each configuration before taking an action to show a warning. Also, shouldn't we do this in `SparkContext` too?

How about we add a new section in  https://spark.apache.org/docs/latest/configuration.html  to collect and show configuration's usage scope. 
Also we can add a `scope` tag in ConfigBuilder for these special configuration. 

Then in this pr, we can just check the config type and warn message.

Reply:
I think you can just add some docs for each configuration like from `blah blah` to `(Launcher scope) blah blah` or something like this.

Reply:
> I think you can just add some docs for each configuration like from `blah blah` to `(Launcher scope) blah blah` or something like this.

Updated, how about current? 

Warn message like 
```

11:38:42.540 WARN org.apache.spark.sql.SparkSession$Builder: Since spark has been submitted, such configurations
 `spark.driver.memory -> 1g` may not take effect.
 For how to set these configuration correctly, you can refer to
 https://spark.apache.org/docs/latest/configuration.html#dynamically-loading-spark-properties.

```